ruy/CMakeLists.txt

```diff
@@ -0,0 +1,1698 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Do not edit! To regenerate, run:
+#   cmake/bazel_to_cmake.sh
+
+if(CMAKE_SYSTEM_NAME STREQUAL Windows)
+  set(ruy_0_Wall_Wcxx14_compat_Wextra_Wundef "")
+else()
+  set(ruy_0_Wall_Wcxx14_compat_Wextra_Wundef "-Wall;-Wextra;-Wc++14-compat;-Wundef")
+endif()
+
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL arm)
+  set(ruy_1_mfpu_neon "-mfpu=neon")
+else()
+  set(ruy_1_mfpu_neon "")
+endif()
+
+if(TRUE)
+  set(ruy_2_O3 "")
+else()
+  set(ruy_2_O3 "-O3")
+endif()
+
+ruy_cc_library(
+  NAME
+    ruy_platform
+  HDRS
+    platform.h
+  COPTS
+    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_1_mfpu_neon}
+    ${ruy_2_O3}
+)
+
+ruy_cc_library(
+  NAME
+    ruy_gtest_wrapper
+  TESTONLY
+  HDRS
+    gtest_wrapper.h
+  DEPS
+    gtest
+)
+
+ruy_cc_library(
+  NAME
+    ruy_check_macros
+  HDRS
+    check_macros.h
+  COPTS
+    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_1_mfpu_neon}
+    ${ruy_2_O3}
+)
+
+ruy_cc_test(
+  NAME
+    ruy_check_macros_test
+  SRCS
+    check_macros_test.cc
+  COPTS
+    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_1_mfpu_neon}
+    ${ruy_2_O3}
+  DEPS
+    ruy_check_macros
+    ruy_gtest_wrapper
+)
+
+ruy_cc_library(
+  NAME
+    ruy_opt_set
+  HDRS
+    opt_set.h
+  COPTS
+    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
```

Comment:
Seems like you're aiming for unique names here. You might want to have the indexing not reset between rules

Reply:
The sharing across rules is intentional. The unique index (`ruy_N_...`) is here to prevent two similar-but-not-identical selects within the same package from getting mixed up.  The descriptive part of the name (here `Wall_Wcxx14_compat_Wextra_Wundef`) is just an aid to humans reading this file, it doesn't play a functional role. The package prefix (here `ruy_`) prevents mixups across packages (as the index gets reset in each package).